src/shell-integration/bash/ghostty.bash

```diff
@@ -47,7 +47,7 @@ if [ -n "$GHOSTTY_BASH_INJECT" ]; then
     if builtin shopt -q login_shell; then
       if [[ $__ghostty_bash_flags != *"--noprofile"* ]]; then
         [ -r /etc/profile ] && builtin source "/etc/profile"
-        for rcfile in ~/.bash_profile ~/.bash_login ~/.profile; do
+        for rcfile in "$HOME/.bash_profile" "$HOME/.bash_login" "$HOME/.profile"; do
```

Comment:
This has the advantage that it'll behave correctly if $HOME includes whitespace. Of course it's a niche case, but it's better to be resilient.